### DISCUSSION TITLE
How do I implement pure MSC device, not composite device?

### DISCUSSION THREAD

**Discussion by skobkars** [hathach/tinyusb#1566] [Q&A]
I need to get files from a device that apparently does not deal with composite USB devices, and needs a simple Mass Storage device to be connected.

Is there a simple way of doing that?

I am working with Adafruit Feather ESP32-S2 under Arduino IDE through Adafruit_TinyUSB library.

Thank you,

**Comment by hathach** [comment]
call Serial.end() then add your msc.begin()

https://github.com/adafruit/Adafruit_TinyUSB_Arduino/blob/master/examples/CDC/no_serial/no_serial.ino

It will most on most port, but may not work on S2, it may not work due to how espressif implementation of descriptors. IRRC, most of the descriptors are pre-determined in initialization (before setup()). Though if you use their built-in implementation instead of Adafruit_TinyUSB, using menu option cdc=off, msc=on may work. In that case, you should post this in their forum support.

**Comment by skobkars** [comment]
Yes I tried that, but with Adafruit implementation my ESP32-S2 still shows as a composite type, though with only MSC child, no serial. The device I am trying to work with does not support composite devices unfortunately.

I moved on to [EspTinyUSB](https://github.com/chegewara/EspTinyUSB) and added SdFat 2.0 support to it instead. It works with both enabled and disabled on-boot CDC option for the ESP32 cores.

Thanks.
